vng_api_common/polymorphism.py

```diff
@@ -190,3 +194,37 @@ def to_internal_value(self, data):
         if extra_fields:
             internal_value.update(extra_fields)
         return internal_value
+
+
+class PolymorphicSerializerExtension(OpenApiSerializerExtension):
+    target_class = "vng_api_common.polymorphism.PolymorphicSerializer"
+    match_subclasses = True
+
+    def map_serializer(self, auto_schema, direction):
+        serializer = self.target
+        schema = auto_schema._map_basic_serializer(serializer, direction)
+        root_component = auto_schema.resolve_serializer(serializer, direction)
+
+        for attr, model_serializer in serializer.discriminator.mapping.items():
+            linked_schema = {"allOf": [root_component.ref]}
+
+            if model_serializer:
+                component = auto_schema.resolve_serializer(model_serializer, direction)
+
+                if component:
+                    linked_schema["allOf"].append(component.ref)
+
+            linked_component = ResolvedComponent(
+                name=attr, type=ResolvedComponent.SCHEMA, schema=linked_schema
+            )
+            auto_schema.registry.register_on_missing(linked_component)
+
+        polymorphic_schema = {
+            "discriminator": {
+                "propertyName": underscore_to_camel(
+                    serializer.discriminator.discriminator_field
+                )
+            }
+        }
+
+        return {**schema, **polymorphic_schema}
```

Comment:
Deze extension [zit er al in](https://github.com/MatthijsBekendam/vng-api-common/blob/drf_compatibilty_newer_django_fix/vng_api_common/inspectors/polymorphic.py#L7)